src/transaction/transaction.rs

```diff
@@ -644,6 +672,10 @@ impl Transaction {
             .retry_region(self.options.retry_options.region_backoff.clone())
             .merge(Collect)
             .plan();
+        if !self.is_heartbeat_started && self.options.auto_heartbeat {
+            self.is_heartbeat_started = true;
+            self.start_auto_heartbeat().await;
+        }
```

Comment:
I think we should execute the lock plan before we start the heartbeat - we don't want the heartbeat to arrive before the write.